airbyte-integrations/connectors/destination-milvus/destination_milvus/indexer.py

```diff
@@ -94,6 +95,18 @@ def _delete_for_filter(self, expr: str) -> None:
             self._collection.delete(expr=f"{id_field} in [{id_list_expr}]")
             page = iterator.next()
 
+    def _normalize(self, metadata: dict) -> dict:
+        result = {}
+
+        for key, value in metadata.items():
+            normalized_key = key
+            # the primary key can't be set directly with auto_id, so we prefix it with an underscore
+            if key == self._primary_key:
+                normalized_key = f"_{key}"
+            result[normalized_key] = value
```

Comment:
The Milvus collection has to be set to auto_id, but if that's the case Milvus will complain if an entity contains a value for the primary key (as it will pick a value itself). So let's say you create a collection with primary key `id` set to auto_id, but a record coming from some source contains an `id` field as well, it will error.

This field changes the logic so the `id` field from the Airbyte record is stored as `_id` in the collection. This is necessary because one record might be split into multiple entities, so `_id` won't be guaranteed to be unique.